subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/events/DefaultTransformDescriptor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.provider.events;
+
+import org.gradle.tooling.internal.protocol.events.InternalTransformDescriptor;
+
+import java.util.Set;
+
+public class DefaultTransformDescriptor extends DefaultOperationDescriptor implements InternalTransformDescriptor {
+
+    private final String transformerName;
+    private final String subjectName;
+    private final Set<InternalTransformDescriptor> dependencies;
+
+    public DefaultTransformDescriptor(Object id, String displayName, Object parentId, String transformerName, String subjectName, Set<InternalTransformDescriptor> dependencies) {
+        super(id, displayName, displayName, parentId);
+        this.transformerName = transformerName;
+        this.subjectName = subjectName;
+        this.dependencies = dependencies;
+    }
+
+    @Override
+    public String getTransformerName() {
+        return transformerName;
+    }
+
+    @Override
+    public String getSubjectName() {
+        return subjectName;
+    }
+
+    @Override
+    public Set<InternalTransformDescriptor> getDependencies() {
```

Comment:
Can't a transformation also depend on tasks?

Reply:
@lptr @ljacomet Can it?

Reply:
I'm pretty sure, since transforms can run on project dependencies and those need to be built before the transform can run.

Reply:
I confirm this. Also as we are introducing the ability for a transform to see the artifacts from the dependencies of the artifact it is transforming, that's another form of task dependencies appearing here when the artifact is a project.

Reply:
@ljacomet Thanks! Can you point me to a test case that shows how to set this up?

Reply:
Have a look at `ArtifactTransformWithDependenciesIntegrationTest`

Reply:
Thanks, added in e85fb9a33b2.